### PR TITLE
O11Y-2370: Report Code Coverage on PRs

### PR DESCRIPTION
**Pull request by dustinlessard-wf**
## Which problem is this PR solving?

It is often very useful to be aware of code coverage at the time one is introducing changes.  We should leverage the Github action that will take the coverage report and comment on the pull request.

## Reviewers

@Workiva/observability @evanweible-wf

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by semveraudit-wf** [comment]
### Public API Changes

 No changes to the public API found for commit b200f12e76078f01b70030314d935f4b87d2171b

Showing results for b200f12e76078f01b70030314d935f4b87d2171b
> Powered by [semver-audit-service](https://github.com/Workiva/semver-audit-service). Please report any problems by [filing an issue](https://github.com/Workiva/semver-audit-service/issues).
> Reported by the dart semver audit client 2.2.2
[Browse public API](https://semver-audit.wk-dev.wdesk.org/Workiva/opentelemetry-dart/b200f12e76078f01b70030314d935f4b87d2171b).

File: .github/workflows/dart_ci.yaml [hunk 1]
```diff
@@ -34,3 +38,13 @@ jobs:
         run: make init
       - name: Format, analyze, and run tests
         run: make test
+      - name: Generate Coverage
+        run: dart run test --coverage=./coverage
```
Comment (blakeroberts-wk) [inline .github/workflows/dart_ci.yaml:42]: is it possible to ignore certain files? we might want to ignore the generated pb files

**Review by blakeroberts-wk — COMMENTED** [review]
one change if possible, else lgtm

File: .github/workflows/dart_ci.yaml (hunk 1 above)
Comment (dustinlessard-wf) [reply .github/workflows/dart_ci.yaml:42]: ^^ Yeah, we totally should do that.  I'm still figuring out the right regex.

Comment (dustinlessard-wf) [reply .github/workflows/dart_ci.yaml:42]: It's going to require the use of another library to process the coverage file and remove the files we don't care about.  I'd like to suggest we move ahead with this pr and follow up with that next step.

Comment (blakeroberts-wk) [reply .github/workflows/dart_ci.yaml:42]: Quick googling, it seems like the easiest option might be to use a different lcov gh action such as https://github.com/marketplace/actions/very-good-coverage

Comment (dustinlessard-wf) [reply .github/workflows/dart_ci.yaml:42]: The goal of this pr is to ensure that code coverage is reported on the pull request.  The 'very-good-coverage' doesn't comment on the pr.

**Comment by dustinlessard-wf** [comment]
FYI @blakeroberts-wk , generated files are now ignored

**Review by blakeroberts-wk — APPROVED** [review]

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**